src/pages/index.js

```diff
@@ -58,48 +26,159 @@ const editModalDescriptionInput = editModal.querySelector(
+
+// Preview image elements
 const previewModal = document.querySelector("#preview-modal");
 const previewModalImageEl = previewModal.querySelector(".modal__image");
 const previewModalCaptionEl = previewModal.querySelector(".modal__caption");
 const previewModalCloseBtn = previewModal.querySelector(
   ".modal__close-btn_type_preview"
 );
 
-const closeButtons = document.querySelectorAll(".modal__close-btn");
+// const closeButtons = document.querySelectorAll(".modal__close-btn");
 
 const cardTemplate = document.querySelector("#card-template");
 const cardsList = document.querySelector(".cards__list");
 
-const logoImage = document.getElementById("image-logo");
-logoImage.src = logoMain;
+const profileAvatar = document.querySelector(".profile__avatar");
+const avatarEl = document.getElementById("avatar-id");
+avatarEl.src = avatarSrc;
 
-const pencilImage = document.getElementById("image-pencil");
-pencilImage.src = pencilSrc;
+const api = new Api({
+  baseUrl: "https://around-api.en.tripleten-services.com/v1",
+  headers: {
+    authorization: "27d33ac4-8679-4bf4-8da5-2cc7ff6ae6cc",
+    "Content-Type": "application/json",
+  },
+});
 
-const plusImage = document.getElementById("image-plus");
-plusImage.src = plusSrc;
+api
+  .getAppInfo()
+  .then(([userInfo, cards]) => {
+    profileName.textContent = userInfo.name;
+    profileDescription.textContent = userInfo.about;
+    profileAvatar.src = userInfo.avatar;
+    cards.forEach((item) => {
+      renderCard(item, "append");
+    });
+  })
+  .catch(console.error);
 
 function handleEditFormSubmit(evt) {
   evt.preventDefault();
-  profileName.textContent = editModalNameInput.value;
-  profileDescription.textContent = editModalDescriptionInput.value;
-  closeModal(editModal);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true);
+  api
+    .editUserInfo({
+      name: editModalNameInput.value,
+      about: editModalDescriptionInput.value,
+    })
+    .then((data) => {
+      profileName.textContent = data.name;
+      profileDescription.textContent = data.about;
+      closeModal(editModal);
+    })
+    .catch(console.error)
+    .finally(() => {
+      setButtonText(submitBtn, false);
+    });
 }
 
 function handleAddCardSubmit(evt) {
   evt.preventDefault();
-  const inputValues = { name: cardNameInput.value, link: cardLinkInput.value };
-  renderCard(inputValues);
-  cardForm.reset();
-  disableButton(cardSubmitButton, settings);
-  closeModal(cardModal);
+  const submitBtn = evt.submitter;
+  setButtonText(submitBtn, true, "Save", "Saving...");
+  submitBtn.disabled = true;
+
+  api
+    .newCardInfo({ name: cardNameInput.value, link: cardLinkInput.value })
+    .then((newCard) => {
+      renderCard(newCard, "prepend");
+      closeModal(cardModal);
+      cardForm.reset();
+
+      const inputList = Array.from(
+        cardForm.querySelectorAll(settings.inputSelector)
+      );
+      resetValidation(cardForm, inputList, settings);
```

Comment:
`resetValidation` is for clearing possible validation errors when you open the **profile** form, because you insert **valid** data into its inputs. But when you add a new card, you don't have any errors after submitting. You need to call `disableButton` there.



```js
disableButton(submitBtn, settings);
```